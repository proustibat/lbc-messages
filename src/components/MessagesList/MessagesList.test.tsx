import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MessagesList } from './index';
import { messages } from '../../fixtures/messages';

describe('MessagesList', () => {
  afterEach(cleanup);

  it('should render an empty message information and no message component', () => {
    // Given / When
    const { getByTestId, queryAllByTestId } = render(
      <MessagesList messages={[]} />,
    );

    // Then
    expect(getByTestId('counter').textContent).toBe('empty-list');
    expect(queryAllByTestId('message-accordion')).toHaveLength(0)
  });

  it('should render the messages', () => {
    // Given / When
    const { getAllByTestId } = render(<MessagesList messages={messages} />);

    // Then
    expect(getAllByTestId('message-accordion')).toHaveLength(messages.length)
    expect(getAllByTestId('message-accordion')).toMatchSnapshot();
  });

  it('should display the number of messages', () => {
    // Given / When
    const { getByTestId } = render(<MessagesList messages={messages} />);

    // Then
    expect(getByTestId('counter').textContent).toBe('messageWithCounter');
  });
});
