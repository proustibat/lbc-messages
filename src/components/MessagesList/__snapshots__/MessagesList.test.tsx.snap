// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList should render the messages 1`] = `
Array [
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="message-accordion"
  >
    <div
      aria-controls="111-content"
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      id="111-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          />
        </svg>
        <p
          class="MuiTypography-root heading_fontSize-0.9375rem heading_marginLeft-8 MuiTypography-body1"
          data-testid="title"
        >
          title 1
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="111-header"
            id="111-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <p
                class="MuiTypography-root body_fontSize-0.9375rem body_color-rgba(0, 0, 0, 0.54) MuiTypography-body1"
              >
                body 1
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="message-accordion"
  >
    <div
      aria-controls="222-content"
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      id="222-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        <p
          class="MuiTypography-root heading_fontSize-0.9375rem heading_marginLeft-8 MuiTypography-body1"
          data-testid="title"
        >
          title 2
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="222-header"
            id="222-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <p
                class="MuiTypography-root body_fontSize-0.9375rem body_color-rgba(0, 0, 0, 0.54) MuiTypography-body1"
              >
                body 2
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="message-accordion"
  >
    <div
      aria-controls="333-content"
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      id="333-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        <p
          class="MuiTypography-root heading_fontSize-0.9375rem heading_marginLeft-8 MuiTypography-body1"
          data-testid="title"
        >
          title 3
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="333-header"
            id="333-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <p
                class="MuiTypography-root body_fontSize-0.9375rem body_color-rgba(0, 0, 0, 0.54) MuiTypography-body1"
              >
                body 3
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
